pkg/vpcmodel/semanticDiffSubnets.go

```diff
@@ -213,10 +188,325 @@ func diffDescription(diff DiffType) string {
+// Two equivalent SubnetConnectivityMap objects s.t. any (src1, dst1) of subnet1Connectivity and
+// (src2, dst2) of subnet2Connectivity s.t. if src1 and src2 (dst1 and dst2) are both external then
+// they are either equal or disjoint
+func (subnetConfConnectivity *SubnetConfigConnectivity) GetConnectivesWithSameIPBlocks(otherConfConnectivity *SubnetConfigConnectivity) (
+	alignedConnectivityConfig, alignedOtherConnectivityConfig *SubnetConfigConnectivity, myErr error) {
+	// 1. computes new set of external nodes (only type of nodes here) in cfg1 and cfg2
+	// does so by computing disjoint block between src+dst ipBlocks in cfg1 and in cfg2
+	// the new set of external nodes is determined based on them
+	connectivityIPBlist, err := subnetConfConnectivity.subnetConnectivity.getIPBlocksList()
+	if err != nil {
+		return nil, nil, err
+	}
+	otherIPBlist, err := otherConfConnectivity.subnetConnectivity.getIPBlocksList()
+	if err != nil {
+		return nil, nil, err
+	}
+	disjointIPblocks := common.DisjointIPBlocks(connectivityIPBlist, otherIPBlist)
+	// 2. copy configs and generates Nodes[] as per disjointIPblocks
+	alignedConfig, err := subnetConfConnectivity.config.copyConfig(disjointIPblocks)
+	if err != nil {
+		return nil, nil, err
+	}
+	otherAlignedConfig, err := otherConfConnectivity.config.copyConfig(disjointIPblocks)
+	if err != nil {
+		return nil, nil, err
+	}
+	// 3. resize connections as per the new Nodes[]
+	alignedConnectivity, err := subnetConfConnectivity.subnetConnectivity.alignConnectionsGivenIPBlists(
+		alignedConfig, disjointIPblocks)
+	if err != nil {
+		return nil, nil, err
+	}
+	alignedOtherConnectivity, err := otherConfConnectivity.subnetConnectivity.alignConnectionsGivenIPBlists(
+		otherAlignedConfig, disjointIPblocks)
+	if err != nil {
+		return nil, nil, err
+	}
+	return &SubnetConfigConnectivity{alignedConfig, alignedConnectivity},
+		&SubnetConfigConnectivity{otherAlignedConfig, alignedOtherConnectivity}, myErr
+}
+
+func (subnetConnectivity *SubnetConnectivityMap) alignConnectionsGivenIPBlists(config *VPCConfig, disjointIPblocks []*common.IPBlock) (
+	alignedConnectivity SubnetConnectivityMap, err error) {
+	alignedConnectivitySrc, err := subnetConnectivity.actualAlignSrcOrDstGivenIPBlists(config, disjointIPblocks, true)
+	if err != nil {
+		return nil, err
+	}
+	alignedConnectivity, err = alignedConnectivitySrc.actualAlignSrcOrDstGivenIPBlists(config, disjointIPblocks, false)
+	return alignedConnectivity, err
+}
+
+// aligned config: copies from old config everything but nodes,
+// nodes are resized by disjointIPblocks
+func (c *VPCConfig) copyConfig(disjointIPblocks []*common.IPBlock) (alignedConfig *VPCConfig, err error) {
+	// copy config
+	alignedConfig = c
+	//  nodes - external addresses - are resized
+	alignedConfig.Nodes, err = resizeNodes(c.Nodes, disjointIPblocks)
+	return alignedConfig, err
```

Comment:
Yes, this is what it is meant to do.

Reply:
so the function name `copyConfig` is a bit misleading.. maybe change it to something like `refineConfigExternalNodes`.
also, there is no need to return another `*VPCConfig` object, since the function just changes the receiver `c` itself.

Reply:
ok, done